File: FleetMonitoring.Web/Controllers/OwnerController.vb
Imports System.Web.Mvc

Namespace Controllers
    Public Class OwnerController
        Inherits Controller

        ' GET: Owner
        Function Index() As ActionResult
            Return View()
        End Function

        ' GET: Owner/Details/5
        Function Details(ByVal id As Integer) As ActionResult
            Return View()
        End Function

        ' GET: Owner/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: Owner/Create
        <HttpPost()>
        Function Create(ByVal collection As FormCollection) As ActionResult
            Try
                ' TODO: Add insert logic here

                Return RedirectToAction("Index")
            Catch
                Return View()
            End Try
        End Function

        ' GET: Owner/Edit/5
        Function Edit(ByVal id As Integer) As ActionResult
            Return View()
        End Function

        ' POST: Owner/Edit/5
        <HttpPost()>
        Function Edit(ByVal id As Integer, ByVal collection As FormCollection) As ActionResult
            Try
                ' TODO: Add update logic here

                Return RedirectToAction("Index")
            Catch
                Return View()
            End Try
        End Function

        ' GET: Owner/Delete/5
        Function Delete(ByVal id As Integer) As ActionResult
            Return View()
        End Function

        ' POST: Owner/Delete/5
        <HttpPost()>
        Function Delete(ByVal id As Integer, ByVal collection As FormCollection) As ActionResult
            Try
                ' TODO: Add delete logic here

                Return RedirectToAction("Index")
            Catch
                Return View()
            End Try
        End Function
    End Class
End Namespace

File: FleetMonitoring.WebUI/Controllers/OwnerController.cs
using FleetMonitoring.Data.Repositories;
using FleetMonitoring.WebUI.Models;
using FleetMonitoring.WebUI.Services;
using System.Web.Mvc;

namespace FleetMonitoring.WebUI.Controllers
{
    public class OwnerController : Controller
    {
        protected readonly IOwnerRepository _ownerRepository;

        public OwnerController(IOwnerRepository ownerRepository)
        {
            _ownerRepository = ownerRepository;
        }

        // GET: Owner
        public ActionResult Index()
        {
            var owners = _ownerRepository.GetAll();

            ViewData["Owners"] = owners;

            return View();
        }

        // GET: Owner/Create
        public ActionResult Create()
        {
            return View();
        }

        // POST: Owner/Create
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult Create(OwnerModel collection)
        {
            if (!ModelState.IsValid)
            {
                return View();
            }

            _ownerRepository.Save(OwnerService.MappingOwnerModel(collection));

            return RedirectToAction("Index");
        }

        // GET: Owner/Edit/5
        public ActionResult Edit(int id)
        {
            var owner = OwnerService.MappingToView(_ownerRepository.Get(id));

            return View(owner);
        }

        // POST: Owner/Edit/5
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult Edit(int id, OwnerModel collection)
        {
            if (!ModelState.IsValid)
            {
                return View();
            }

            _ownerRepository.Update(id, OwnerService.MappingOwnerModel(collection, true));
            
            return RedirectToAction("Index");
        }

        // POST: Owner/Delete/5
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult Delete(int id)
        {
            _ownerRepository.Delete(id);

            return RedirectToAction("Index");
        }

        protected override void Dispose(bool disposing)
        {
            _ownerRepository.Dispose();
            base.Dispose(disposing);
        }
    }
}
